driver-core/src/main/com/mongodb/internal/connection/AsynchronousSocketChannelStream.java

```diff
@@ -72,22 +76,34 @@ public void open() throws IOException {
     @Override
     public void openAsync(final AsyncCompletionHandler<Void> handler) {
         isTrue("unopened", channel == null);
-        try {
-            channel = AsynchronousSocketChannel.open(group);
-            channel.setOption(StandardSocketOptions.TCP_NODELAY, true);
-            channel.setOption(StandardSocketOptions.SO_KEEPALIVE, settings.isKeepAlive());
-            if (settings.getReceiveBufferSize() > 0) {
-                channel.setOption(StandardSocketOptions.SO_RCVBUF, settings.getReceiveBufferSize());
-            }
-            if (settings.getSendBufferSize() > 0) {
-                channel.setOption(StandardSocketOptions.SO_SNDBUF, settings.getSendBufferSize());
-            }
+        Queue<SocketAddress> socketAddressQueue = new LinkedList<SocketAddress>(serverAddress.getSocketAddresses());
+        initializeSocketChannel(handler, socketAddressQueue);
+    }
+
+    @SuppressWarnings("deprecation")
+    private void initializeSocketChannel(final AsyncCompletionHandler<Void> handler, final Queue<SocketAddress> socketAddressQueue) {
+        if (socketAddressQueue.isEmpty()) {
+            handler.failed(new MongoSocketException("Exception opening socket", serverAddress));
+        } else {
+            SocketAddress socketAddress = socketAddressQueue.poll();
+
+            try {
+                AsynchronousSocketChannel chl = AsynchronousSocketChannel.open(group);
```

Comment:
Note: Can we make `chl` clearer in turns of naming? I understand that `channel` is used by the outer class. I've found with Java if in doubt be verbose as it reduces the cognitive load when reading the code.